components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthenticator.java

```diff
@@ -65,4 +65,11 @@ boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams
      */
     String getClientId(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             oAuthClientAuthnContext) throws OAuthClientAuthnException;
+
+    /**
+     * Retrieve the referred name of the authenticator.
```

Comment:
In which context we assigned these names to the authenticators? 
For an example, If I am gonna write new authenticator, what should I follow to derive the referred name?
Can we add usecase of this method?

Reply:
A few sample scenarios are mentioned in the PR description as well. These references can be found in the OpenID spec [1]
[1] https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata

Reply:
So referred name actually token_endpoint_auth_method?

Reply:
token_endpoint_auth_method is the attribute that we use in DCR to store the client authentication methods to be used for that application. For that attribute as values we give the referred name. Find below an example.
token_endpoint_auth_method="private_key_jwt"